go-controller/pkg/ovn/ovn.go

```diff
@@ -798,3 +801,35 @@ func shouldUpdate(node, oldNode *kapi.Node) (bool, error) {
 
 	return true, nil
 }
+
+// getAddressSetFromCacheLocked returns the named address set, locked, from the
+// cache or nil
+func (oc *Controller) getAddressSetFromCacheLocked(name string) *addressSet {
```

Comment:
Maybe, I don't understand the whole logic, but if all the public API methods of address set can acquire lock before doing anything, why do we need to acquire the lock here?

Reply:
This code was reworked and removed.